app/models/manageiq/providers/azure_stack/inventory/parser/network_manager.rb

```diff
@@ -3,11 +3,59 @@ def parse
     log_header = "Collecting data for EMS : [#{collector.manager.name}] id: [#{collector.manager.id}]"
     $azure_stack_log.info("#{log_header}...")
 
+    cloud_networks
+    network_ports
     security_groups
 
     $azure_stack_log.info("#{log_header}...Complete")
   end
 
+  def cloud_networks
+    collector.networks.each do |network|
+      persister.cloud_networks.build(
+        :name    => network.name,
+        :ems_ref => network.id.downcase
+      )
+      if network.subnets
+        cloud_subnets(network)
+      end
+    end
+  end
+
+  def cloud_subnets(network)
+    network.subnets.each do |subnet|
+      persister.cloud_subnets.build(
+        :name            => subnet.name,
+        :ems_ref         => subnet.id.downcase,
+        :cloud_network   => persister.cloud_networks.lazy_find(network.id.downcase),
```

Comment:
Since this is called right after building the network you can save the lazy_find and pass in the persisted network object directly if you want.
Example:
```ruby
persister_network = persister.cloud_networks.build()
cloud_subnets(persister_network, network.subnets) if network.subnets
```

Reply:
Ah @miha-plesko beat me to it https://github.com/ManageIQ/manageiq-providers-azure_stack/pull/9#discussion_r326563868